src/main/java/carpet/mixins/ExplosionMixin.java

```diff
@@ -58,4 +70,24 @@ private BlockState noBlockCalcsWithNoBLockDamage(World world, BlockPos pos)
         if (CarpetSettings.explosionNoBlockDamage) return Blocks.BEDROCK.getDefaultState();
         return world.getBlockState(pos);
     }
+
+    @Inject(method = "<init>(Lnet/minecraft/world/World;Lnet/minecraft/entity/Entity;DDDFZLnet/minecraft/world/explosion/Explosion$DestructionType;)V",
+            at = @At(value = "RETURN"))
+    private void onExplosionCreated(World world, Entity entity, double x, double y, double z, float power, boolean createFire, Explosion.DestructionType blockDestructionType, CallbackInfo ci)
+    {
+        if (LoggerRegistry.__explosions && ! world.isClient)
+        {
+            eLogger = new ExplosionLogHelper(entity, x, y, z, power, createFire, blockDestructionType);
+        }
+    }
+
+    @Redirect(method = "collectBlocksAndDamageEntities",
+            at = @At(value = "INVOKE", target = "Lnet/minecraft/entity/Entity;setVelocity(Lnet/minecraft/util/math/Vec3d;)V"))
+    private void setVelocityAndUpdateLogging(Entity entity, Vec3d velocity)
```

Comment:
This might not work if optimized explosions are in plus it may collide with lithium explosions. You should add these trackers to carpet optimized code for explosions as well as test with lithium making sure there is no mixin issues with mod compat.

Reply:
I tested and found it capable with lithium but not with carpet tnt optimization. I will have it fixed as soon as possible.

Reply:
cool np